pkg/devfile/parser/reader.go

```diff
@@ -0,0 +1,107 @@
+package parser
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/devfile/library/pkg/testingutil/filesystem"
+	"github.com/devfile/library/pkg/util"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+	k8yaml "sigs.k8s.io/yaml"
+
+	routev1 "github.com/openshift/api/route/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// YamlSrc specifies the src of the yaml in either Path, URL or Data format
+type YamlSrc struct {
+	// Path is a relative or absolute yaml path.
+	Path string
+	// URL is the URL address of the specific yaml.
+	URL string
+	// Data is the yaml content in []byte format.
+	Data []byte
+}
+
+// ReadKubernetesYaml reads a yaml Kubernetes file from either the Path, URL or Data provided.
+// It returns Deployments, Services, Routes resources as the primary Kubernetes resources.
+// Other Kubernetes resources are returned as []byte type. Consumers interested in other Kubernetes resources
+// are expected to Unmarshal it to the struct of the respective resource.
+func ReadKubernetesYaml(src YamlSrc, fs filesystem.Filesystem) ([]appsv1.Deployment, []corev1.Service, []routev1.Route, [][]byte, error) {
```

Comment:
How are we expecting to use this function? consumer directly call?
I'm wondering should this function be called by `generator` when consumer trying to run `getDeployment`/`getService`/`getRoute`?

Reply:
good question, let me ask now in the meeting

Reply:
discussed in the team meeting. to keep it simple, we currently only provide `ReadKubernetesYaml` as a helper function, and tool should extract the content they need from the returned result.